src/com/amazon/corretto/crypto/provider/keygeneratorspi/SecretKeyGenerator.java

```diff
@@ -0,0 +1,55 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider.keygeneratorspi;
+
+import javax.crypto.KeyGeneratorSpi;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SecretKeyGenerator extends KeyGeneratorSpi {
+    private final Supplier<SecureRandom> defaultSecureRandom;
+    private final SecretKeyProperties secretKeyProperties;
+    private Optional<SecureRandom> secureRandom;
+    private int keySize;
+
+    public SecretKeyGenerator(final Supplier<SecureRandom> defaultSecureRandom, final SecretKeyProperties secretKeyProperties) {
+        this.defaultSecureRandom = Objects.requireNonNull(defaultSecureRandom, "defaultSecureRandom cannot be null.");
+        this.secretKeyProperties = Objects.requireNonNull(secretKeyProperties, "secretKeyProperties cannot be null.");
+        this.secureRandom = Optional.empty();
+        this.keySize = secretKeyProperties.defaultKeySize();
+    }
+
+    @Override
+    protected void engineInit(final SecureRandom random) {
+        secureRandom = Optional.ofNullable(random);
+    }
+
+    @Override
+    protected void engineInit(final AlgorithmParameterSpec params, final SecureRandom random) throws InvalidAlgorithmParameterException {
+        throw new InvalidAlgorithmParameterException("SecretKeyGenerator does not support initialization with AlgorithmParameterSpec.");
+    }
+
+    @Override
+    protected void engineInit(final int keySize, final SecureRandom random) {
+        secretKeyProperties.checkKeySizeIsValid(keySize);
+        this.keySize = keySize;
+        this.secureRandom = Optional.ofNullable(random);
+    }
+
+    @Override
+    protected SecretKey engineGenerateKey() {
+        final byte[] keyBytes = new byte[keySize / 8];
+        final SecureRandom srand = secureRandom.orElseGet(defaultSecureRandom);
+        srand.nextBytes(keyBytes);
```

Comment:
This can throw an NPE if defaultSecureRandom is null.

Reply:
It's interesting that `orElseGet` does not check the nullness of return value of the supplier.